packages/devtools_app/lib/src/performance/timeline_flame_chart.dart

```diff
@@ -471,18 +479,75 @@ class TimelineFlameChartState
           colorScheme: colorScheme,
         ),
       ),
-      CustomPaint(
-        painter: SectionLabelPainter(
-          _timelineController.data.eventGroups,
-          zoom: zoom,
-          constraints: constraints,
-          verticalController: verticalController,
-          horizontalController: horizontalController,
-          chartStartInset: widget.startInset,
-          colorScheme: colorScheme,
+      _buildSectionLabels(constraints: constraints),
+    ];
+  }
+
+  Widget _buildSectionLabels({@required BoxConstraints constraints}) {
+    final colorScheme = Theme.of(context).colorScheme;
+    final eventGroups = _timelineController.data.eventGroups;
+
+    final children = <Widget>[];
+    for (int i = 0; i < eventGroups.length; i++) {
+      var topSpacer = 0.0;
+      var bottomSpacer = 0.0;
+      if (i == 0) {
+        // Add spacing to account for timestamps at top of chart.
+        topSpacer +=
+            sectionSpacing * TimelineFlameChart.rowOffsetForTopPadding -
+                rowHeight;
+      }
+      if (i == eventGroups.length - 1) {
+        // Add spacing to account for bottom row of padding.
+        bottomSpacer = rowHeight;
+      }
+
+      final groupName = eventGroups.keys.elementAt(i);
+      final group = eventGroups[groupName];
+      final backgroundColor = alternatingColorForIndex(
+        eventGroups.values.toList().indexOf(group),
+        colorScheme,
+      );
+      final backgroundWithOpacity = Color.fromRGBO(
+        backgroundColor.red,
+        backgroundColor.green,
+        backgroundColor.blue,
+        0.85,
+      );
+      children.addAll([
```

Comment:
why are topSpacer and bottomSpacer separate widgets instead of just padding on the container?

Reply:
fixed.